File: server/src/main/java/net/bytemc/minestom/server/handler/BlockHandlers.java
package net.bytemc.minestom.server.handler;

import net.minestom.server.instance.block.BlockHandler;
import net.minestom.server.tag.Tag;
import net.minestom.server.utils.NamespaceID;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.List;

public final class BlockHandlers {

    public static BlockHandler BANNER_HANDLER = new BlockHandler() {
        @Override
        public @NotNull NamespaceID getNamespaceId() {
            return NamespaceID.from("minecraft:banner");
        }

        @Override
        public @NotNull Collection<Tag<?>> getBlockEntityTags() {
            return List.of(Tag.NBT("Patterns"));
        }
    };

    public static BlockHandler SKULL_HANDLER = new BlockHandler() {
        @Override
        public @NotNull NamespaceID getNamespaceId() {
            return NamespaceID.from("minecraft:skull");
        }

        @Override
        public @NotNull Collection<Tag<?>> getBlockEntityTags() {
            return List.of(Tag.NBT("SkullOwner"));
        }
    };
}

File: server/src/main/java/net/bytemc/minestom/server/handler/BlockHandlers.kt
package net.bytemc.minestom.server.handler

import net.minestom.server.tag.Tag

import net.minestom.server.instance.block.BlockHandler
import net.minestom.server.utils.NamespaceID


object BlockHandlers {

    val BANNER_HELPER: BlockHandler = object : BlockHandler {
        override fun getNamespaceId(): NamespaceID {
            return NamespaceID.from("minecraft:banner")
        }
        override fun getBlockEntityTags(): Collection<Tag<*>> {
            return listOf(Tag.NBT("Patterns"))
        }
    }

    val SKULL_HANDLER: BlockHandler = object : BlockHandler {
        override fun getNamespaceId(): NamespaceID {
            return NamespaceID.from("minecraft:skull")
        }
        override fun getBlockEntityTags(): Collection<Tag<*>> {
            return listOf(Tag.NBT("SkullOwner"))
        }
    }
}